libs/libc/tls/tls_setdtor.c

```diff
@@ -59,18 +56,16 @@
  *
  ****************************************************************************/
 
-int tls_set_dtor(int tlsindex, tls_dtor_t destr)
+int tls_set_dtor(int tlsindex, tls_dtor_t dtor)
 {
-  FAR struct tcb_s *rtcb = this_task();
-  FAR struct task_group_s *group = rtcb->group;
-  irqstate_t flags;
+  FAR struct task_info_s *tinfo = task_get_info();
 
-  DEBUGASSERT(group != NULL);
+  DEBUGASSERT(tinfo != NULL);
   DEBUGASSERT(tlsindex >= 0 && tlsindex < CONFIG_TLS_NELEM);
 
-  flags = spin_lock_irqsave(NULL);
-  group->tg_tlsdestr[tlsindex] = destr;
-  spin_unlock_irqrestore(NULL, flags);
+  sem_wait(&tinfo->ta_tlssem);
```

Comment:
Should _SEM_WAIT be used?  It should be if this could ever be called from within the OS.  If not, then it should not be built if __KERNEL__ is defined.

Reply:
The TLS key relative interface wouldn't be called from OS side, execpt **nx_pthread_exit** by syscall (drop to user space)

Reply:
> The TLS key relative interface wouldn't be called from OS side, execpt **nx_pthread_exit** by syscall (drop to user space)

Then either ifdef __KERNEL__ should be used to specifically exclude access to the kernel or _SEM_WAIT should be used instead of sem_wait() so that if it is called from the OS, nothing bad happens.  I think everything else in libs/libc uses _SEM_WAIT, I suspect.

Reply:
Yes, I got it